colossalai/booster/plugin/gemini_plugin.py

```diff
@@ -391,11 +440,32 @@ def configure(
             # model = nn.SyncBatchNorm.convert_sync_batchnorm(model, None)
 
             # wrap the model with Gemini
-            model = GeminiDDP(model, **self.gemini_config, verbose=self.verbose)
+            self.dp_group = None
+            self.tp_group = None
+            try:
+                dp_size = dist.get_world_size() // self.tp_size
+                assert dp_size > 1, f"The size of the DP group should be greater than 1. Please reduce the TP group size."
+                self.pg_mesh = ProcessGroupMesh(dp_size, self.tp_size)
+                self.dp_group = self.pg_mesh.get_group_along_axis(DP_AXIS)
+                self.tp_group = self.pg_mesh.get_group_along_axis(TP_AXIS)
+                shard_config = ShardConfig(tensor_parallel_process_group = self.tp_group, 
+                                            enable_tensor_parallelism=self.enable_tensor_parallelism,
+                                            enable_all_optimization=self.enable_all_optimization,
+                                            enable_fused_normalization=self.enable_fused_normalization,
+                                            enable_flash_attention=self.enable_flash_attention,
+                                            enable_jit_fused=self.enable_jit_fused,
+                                            enable_sequence_parallelism=self.enable_sequence_parallelism,
+                                            enable_sequence_overlap=self.enable_sequence_overlap)
+                shardformer = ShardFormer(shard_config)
+                model, _ = shardformer.optimize(model)
+            except NotImplementedError as e:
+                print(f"Tensor Parallelism policy for {model.__class__} is not implemented yet\n.")
```

Comment:
We should not catch this error, as we can not handle it.

Reply:
If it is not implemented in Shardformer, the TP strategy will not be used, and a prompt will be given. If this situation is not caught, the tests for models not implemented in Shardformer will fail.

Reply:
updated